### PR TITLE
Enable unit tests for Linux CI builds

### PR DESCRIPTION
**Pull request by Alexander-Wilms**
This was discussed a bit in https://github.com/vcmi/vcmi/issues/2618

In order to run unit tests during CI, the H3 assets are required. In order to make this less problematic, I encrypted the H3 data with GPG and uploaded them here:

https://github.com/Alexander-Wilms/encrypted_heroes_3_data/tree/master

Then I added the encryption key as a Github Actions secret and used it to decrypt the data during each CI run.

image

In theory, one could print the key to stdout during CI with an additional space in between each character to avoid it being censored, but I don't think that is very realistic considering the game costs 10$ at most on GOG.

Since you don't have this encryption key in this repo, the CI will fail. But you can check [this](https://github.com/Alexander-Wilms/vcmi/actions/runs/5917784106/job/16046459791) Github action run in my fork:

image

One of you would have to encrypt his own copy of the data and add it to a repo of the vcmi org.

Before this is possibly merged, I'd squash the commits.

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -42,7 +42,8 @@
             "inherits" : "default-release",
             "hidden": true,
             "cacheVariables": {
-                "CMAKE_INSTALL_PREFIX" : "/usr/local"
+                "CMAKE_INSTALL_PREFIX" : "/usr/local",
+                "MAKEFLAGS": "-j$(nproc)"
```
Comment (kambala-decapitator) [inline CMakePresets.json:46]: not sure what this is for: we use ninja be default, it already utilizes all cores

**Comment by nullkiller** [comment]
This also opens a door to run some automated AI tests

**Comment by Alexander-Wilms** [comment]
Should the tests be run with the clang and the gcc build?

It adds only 2 min and might catch compiler-specific bugs.

**Comment by nullkiller** [comment]
Btw have you removed video.vid from heroes 3 data? It is big and not needed for tests.

**Comment by Alexander-Wilms** [comment]
Not yet, will do.

**Comment by rilian-la-te** [comment]
@IvanSavenko just TXT files AFAIK. All assets except TXT used only by client. 

I think we can substitute TXT files by appropriately formatted Lorem Ipsum.

**Comment by IvanSavenko** [comment]
> All assets except TXT used only by client.

There are also msk/msg files that might be required for load. They are used on server, but not sure about tests or whether they can be skipped.

> I think we can substitute TXT files by appropriately formatted Lorem Ipsum.

Some txt files require certain content, e.g. creature / artifact / spells / etc stats.

**Comment by rilian-la-te** [comment]
@IvanSavenko digits is not copyrighted, so if we substitute all text files as Lorem ipsum, but retain the digits, I guess we can use that files for tests and lose a copyright. 

About msg files - need check. But tests for now load only lib, not client or server.

**Comment by Alexander-Wilms** [comment]
@nullkiller Would H3Bitmap.lod suffice for the AI tests?

Wouldn't testing using the original file or at least the archive with only text files be better for unit testing, since the test data exactly matches what players will use?

Also, just using the original archive is more straightforward, what would be the disadvantage? That the assets are copyrighted?

**Comment by Alexander-Wilms** [comment]
I extracted H3bitmap.lof using [mmarch](https://github.com/might-and-magic/mmarch) and created a new one containing just the .txt files.

However, the Test stage now get stuck and doesn't print any output:

https://github.com/Alexander-Wilms/vcmi/actions/runs/6215578384/job/16868629227

image

**Comment by rilian-la-te** [comment]
Can you run this locally (with just txt files)?

**Comment by Alexander-Wilms** [comment]
Strangely, I get 57 failed tests with bot the original .lod and the minmal .lod:

[only_text_data.log](https://github.com/vcmi/vcmi/files/12643889/only_text_data.log)
[original_data.log](https://github.com/vcmi/vcmi/files/12643890/original_data.log)

Do the tests (except for 2) still succeed when you run them?

**Comment by Alexander-Wilms** [comment]
Now I got the tests to run during CI again (using the minimal .lod):

```
77% tests passed, 63 tests failed out of 275

Total Test time (real) =  73.56 sec

The following tests FAILED:
	  8 - BattleFinishedTest.NoBattleIsDraw (SEGFAULT)
	 82 - CGameStateTest.issue2765 (SEGFAULT)
	 83 - CGameStateTest.battleResurrection (SEGFAULT)
	 87 - MapFormat.Random (Failed)
	 88 - MapFormat.Objects (Failed)
	 89 - MapFormat.Terrain (Failed)
	113 - CatapultTest.ApplicableInTown (Failed)
	114 - CatapultApplyTest.DamageToIntactPart (Failed)
	118 - CloneApplyTest.AddsNewUnit (SEGFAULT)
	119 - CloneApplyTest.SetsClonedFlag (SEGFAULT)
	120 - CloneApplyTest.SetsCloneLink (SEGFAULT)
	121 - CloneApplyTest.SetsLifetimeMarker (SEGFAULT)
	124 - DamageApplyTest.DoesDamageToAliveUnit (SEGFAULT)
	125 - DamageApplyTest.IgnoresDeadUnit (SEGFAULT)
	126 - DamageApplyTest.DoesDamageByPercent (SEGFAULT)
	127 - DamageApplyTest.DoesDamageByCount (SEGFAULT)
	128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Failed)
```

https://github.com/Alexander-Wilms/vcmi/actions/runs/6216442472/job/16870299367

**Comment by Alexander-Wilms** [comment]
With ffaf4c5b0104195a6f649f23f12c7386684d1532 I get the following result on my machine:

```
$ git rev-parse HEAD
ffaf4c5b0104195a6f649f23f12c7386684d1532
$ cmake --build -j4 --preset linux-gcc-test
[442/442] Linking CXX executable bin/vcmitest
$ ctest --preset linux-gcc-test
[...]
79% tests passed, 56 tests failed out of 261

Total Test time (real) =  70.17 sec

The following tests FAILED:
         82 - CGameStateTest.issue2765 (SEGFAULT)
         83 - CGameStateTest.battleResurrection (SEGFAULT)
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
         89 - MapFormat.Terrain (Failed)
        113 - CatapultTest.ApplicableInTown (Failed)
        114 - CatapultApplyTest.DamageToIntactPart (Failed)
        118 - CloneApplyTest.AddsNewUnit (SEGFAULT)
        119 - CloneApplyTest.SetsClonedFlag (SEGFAULT)
        120 - CloneApplyTest.SetsCloneLink (SEGFAULT)
        121 - CloneApplyTest.SetsLifetimeMarker (SEGFAULT)
        124 - DamageApplyTest.DoesDamageToAliveUnit (SEGFAULT)
        125 - DamageApplyTest.IgnoresDeadUnit (SEGFAULT)
        126 - DamageApplyTest.DoesDamageByPercent (SEGFAULT)
        127 - DamageApplyTest.DoesDamageByCount (SEGFAULT)
        128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Failed)
        129 - DispelTest.IgnoresOwnEffects (Failed)
        132 - DispelApplyTest.RemovesEffects (SEGFAULT)
        139 - HealTest.NotApplicableIfDeadUnitIsBlocked (Failed)
        140 - HealTest.ApplicableWithAnotherDeadUnitInSamePosition (Failed)
        143 - SacrificeApplyTest.ResurrectsTarget (SEGFAULT)
        163 - ByConfig1/HealApplyTest.Heals/(1-byte object <00>, 1-byte object <01>) (SEGFAULT)
        164 - ByConfig1/HealApplyTest.Heals/(1-byte object <01>, 1-byte object <01>) (SEGFAULT)
        165 - ByConfig1/HealApplyTest.Heals/(1-byte object <02>, 1-byte object <01>) (SEGFAULT)
        166 - ByConfig2/HealApplyTest.Heals/(1-byte object <01>, 1-byte object <00>) (SEGFAULT)
        167 - ByConfig2/HealApplyTest.Heals/(1-byte object <02>, 1-byte object <00>) (SEGFAULT)
        170 - ByConfig/SummonTest.Applicable/(-1, true, false) (Failed)
        171 - ByConfig/SummonTest.Applicable/(-1, true, true) (Failed)
        174 - ByConfig/SummonTest.Applicable/(112, true, false) (Failed)
        175 - ByConfig/SummonTest.Applicable/(112, true, true) (Failed)
        178 - ByConfig/SummonTest.Applicable/(114, true, false) (Failed)
        179 - ByConfig/SummonTest.Applicable/(114, true, true) (Failed)
        180 - ByConfig/SummonTest.Transform/(-1, false, false) (Failed)
        181 - ByConfig/SummonTest.Transform/(-1, false, true) (Failed)
        182 - ByConfig/SummonTest.Transform/(-1, true, false) (Failed)
        183 - ByConfig/SummonTest.Transform/(-1, true, true) (Failed)
        184 - ByConfig/SummonTest.Transform/(112, false, false) (Failed)
        185 - ByConfig/SummonTest.Transform/(112, false, true) (Failed)
        186 - ByConfig/SummonTest.Transform/(112, true, false) (Failed)
        187 - ByConfig/SummonTest.Transform/(112, true, true) (Failed)
        188 - ByConfig/SummonTest.Transform/(114, false, false) (Failed)
        189 - ByConfig/SummonTest.Transform/(114, false, true) (Failed)
        190 - ByConfig/SummonTest.Transform/(114, true, false) (Failed)
        191 - ByConfig/SummonTest.Transform/(114, true, true) (Failed)
        192 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, false) (SEGFAULT)
        193 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, true) (SEGFAULT)
        194 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, false) (SEGFAULT)
        195 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, true) (SEGFAULT)
        196 - ByConfig/SummonApplyTest.UpdatesOldUnit/(false, false) (SEGFAULT)
        197 - ByConfig/SummonApplyTest.UpdatesOldUnit/(false, true) (SEGFAULT)
        198 - ByConfig/SummonApplyTest.UpdatesOldUnit/(true, false) (SEGFAULT)
        199 - ByConfig/SummonApplyTest.UpdatesOldUnit/(true, true) (SEGFAULT)
        200 - ByConfig/TimedApplyTest.ChangesBonuses/(false, false) (SEGFAULT)
        201 - ByConfig/TimedApplyTest.ChangesBonuses/(false, true) (SEGFAULT)
        202 - ByConfig/TimedApplyTest.ChangesBonuses/(true, false) (SEGFAULT)
        203 - ByConfig/TimedApplyTest.ChangesBonuses/(true, true) (SEGFAULT)
```

**Comment by Alexander-Wilms** [comment]
With 4d05023412024659644da531bec0ddd160747cbe (the commit of the CI run linked in the very first comment) I get the following result on my machine:

```
$ git rev-parse HEAD
4d05023412024659644da531bec0ddd160747cbe
$ cmake --build -j4 --preset linux-gcc-test
[222/396] Building CXX object test/CMakeFiles/vcmitest.dir/battle/CBattleInfoCallbackTest.cpp.o
In file included from /home/alexander/Technik/Development/vcmi/test/battle/../../lib/battle/ReachabilityInfo.h:12,
                 from /home/alexander/Technik/Development/vcmi/test/battle/../../lib/battle/CBattleInfoCallback.h:15,
                 from /home/alexander/Technik/Development/vcmi/test/battle/CBattleInfoCallbackTest.cpp:12:
/home/alexander/Technik/Development/vcmi/test/battle/../../lib/battle/CBattleInfoEssentials.h:37:19: warning: defaulted move assignment for ‘CBattleInfoEssentials’ calls a non-trivial move assignment operator for virtual base ‘CCallbackBase’ [-Wvirtual-move-assign]
   37 | class DLL_LINKAGE CBattleInfoEssentials : public virtual CCallbackBase, public IBattleInfoCallback
      |                   ^~~~~~~~~~~~~~~~~~~~~
/home/alexander/Technik/Development/vcmi/test/battle/../../lib/battle/CBattleInfoCallback.h:53:19: warning: defaulted move assignment for ‘CBattleInfoCallback’ calls a non-trivial move assignment operator for virtual base ‘CBattleInfoEssentials’ [-Wvirtual-move-assign]
   53 | class DLL_LINKAGE CBattleInfoCallback : public virtual CBattleInfoEssentials
      |                   ^~~~~~~~~~~~~~~~~~~
[396/396] Linking CXX executable bin/vcmitest
$ ctest --preset linux-gcc-test
[...]
99% tests passed, 2 tests failed out of 259

Total Test time (real) =  68.51 sec

The following tests FAILED:
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
Errors while running CTest
Output from these tests are in: /home/alexander/Technik/Development/vcmi/out/build/linux-gcc-test/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

So something must've broken them in between. I would be good if someone could bisect this.

**Comment by Alexander-Wilms** [comment]
b77bccdc24be8f77f924e6124b35be3407167e37

```
97% tests passed, 7 tests failed out of 261

Total Test time (real) =  69.25 sec

The following tests FAILED:
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
        128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Failed)
        129 - DispelTest.IgnoresOwnEffects (Failed)
        132 - DispelApplyTest.RemovesEffects (Failed)
        193 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, true) (SEGFAULT)
        195 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, true) (SEGFAULT)
Errors while running CTest
Output from these tests are in: /home/alexander/Technik/Development/vcmi/out/build/linux-gcc-test/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

**Comment by Alexander-Wilms** [comment]
65884b15d0895fa5af3c000a2fa6c8d62ba0f5cb

```
99% tests passed, 2 tests failed out of 259

Total Test time (real) =  62.30 sec

The following tests FAILED:
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
Errors while running CTest
Output from these tests are in: /home/alexander/Technik/Development/vcmi/out/build/linux-gcc-test/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
```

**Comment by Alexander-Wilms** [comment]
```
97% tests passed, 7 tests failed out of 261

Total Test time (real) =  79.09 sec

The following tests FAILED:
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
        128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Failed)
        129 - DispelTest.IgnoresOwnEffects (Failed)
        132 - DispelApplyTest.RemovesEffects (Failed)
        193 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, true) (SEGFAULT)
        195 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, true) (SEGFAULT)
Errors while running CTest
Output from these tests are in: /home/alexander/Technik/Development/vcmi/out/build/linux-gcc-test/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
alexander@PC:~/Technik/Development/vcmi> git rev-parse HEAD
4b807e01f1e4d490f315374468a26356d873de40
```

**Comment by Alexander-Wilms** [comment]
97% tests passed, 7 tests failed out of 261

Total Test time (real) =  68.47 sec

The following tests FAILED:
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
        128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Failed)
        129 - DispelTest.IgnoresOwnEffects (Failed)
        132 - DispelApplyTest.RemovesEffects (Failed)
        193 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, true) (SEGFAULT)
        195 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, true) (SEGFAULT)
Errors while running CTest
Output from these tests are in: /home/alexander/Technik/Development/vcmi/out/build/linux-gcc-test/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
alexander@PC:~/Technik/Development/vcmi> git rev-parse HEAD
83b00bb7c3be88015134bc4533c3d15cb80b290e

**Comment by Alexander-Wilms** [comment]
```
git bisect bad 83b00bb7c3be88015134bc4533c3d15cb80b290e
The merge base 83b00bb7c3be88015134bc4533c3d15cb80b290e is bad.
This means the bug has been fixed between 83b00bb7c3be88015134bc4533c3d15cb80b290e and [cfe87e46a92f90dcfbef487e2ce3b15a86d4eebe].
```

**Comment by Alexander-Wilms** [comment]
```
99% tests passed, 2 tests failed out of 259

Total Test time (real) =  64.64 sec

The following tests FAILED:
         87 - MapFormat.Random (Failed)
         88 - MapFormat.Objects (Failed)
Errors while running CTest
Output from these tests are in: /home/alexander/Technik/Development/vcmi/out/build/linux-clang-test/Testing/Temporary/LastTest.log
Use "--rerun-failed --output-on-failure" to re-run the failed cases verbosely.
alexander@PC:~/Technik/Development/vcmi> git rev-parse HEAD
f13a53c1d9baba682bd56e9b1a292d47c7aca2c8
```

**Comment by Alexander-Wilms** [comment]
So, at last I made some progress. 4b807e01f1e4d490f315374468a26356d873de40 seems to have broken these tests:

        128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Failed)
        129 - DispelTest.IgnoresOwnEffects (Failed)
        132 - DispelApplyTest.RemovesEffects (Failed)
        193 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, true) (SEGFAULT)
        195 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, true) (SEGFAULT)
        
`git show 4b1224ec8ccf05d2d228a19e3dcc9c6e1c6e4e44^1` shows commit f13a53c1d9baba682bd56e9b1a292d47c7aca2c8, which as only 2 failed tests.

Could this PR be merged before all tests are fixed or not?

**Comment by nullkiller** [comment]
Could we just disable the failing tests and merge it? I think it is better then leaving it in the state it is now. We can create an issue regarding fixing those failing tests.

**Comment by nullkiller** [comment]
At least I can consider merging it myself as soon as it gets green build if there is no objection.

**Comment by nullkiller** [comment]
This PR is interesting for me as it opens possibility to run headless tests with real maps. But it will need full h3 files available and may take a lot of time so should not be a part of PR check but some nightly/weekly job or something

**Comment by Alexander-Wilms** [comment]
@nullkiller Could you upload the encrypted H3 data to a VCMI repo and share the variable name of the GitHub secret?

And should the failing tests be disabled in this PR?

**Comment by nullkiller** [comment]
Write in slack what exactly needs to be done to upload things, please and we will discuss there. We need to find a good place for it. Failing tests should be disabled if you do not want to fix them. I see you already did a good job.

**Comment by Alexander-Wilms** [comment]
I get this error when I try to sign up via slack.vcmi.eu: "Something went wrong, try again!"

But I can explain it here:

You'll need to create another repo in the vcmi org, clone it and create a directory `HoMM3_data` within in. Then copy the Data and Maps folders into that directory, delete Data/VIDEO.VID, copy the crypt.py script from https://github.com/Alexander-Wilms/encrypted_heroes_3_data into the root dir of the repo and execute it with `./crypt.py -p "password" -e`. Then commit and push.

Then open https://github.com/vcmi/vcmi/settings/secrets/actions/new and create a secret named `HEROES_3_DATA_PASSWORD` containing the password you used.

Can you open a PR against this PR's branch to disable the broken tests?

**Comment by Alexander-Wilms** [comment]
I tried

```
gtest_discover_tests(vcmitest
	WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/bin/"
	EXCLUDE BattleFinishedTest.NoBattleIsDraw)
```

but for some reason `--gtest_also_run_disabled_tests` gets passed to `vcmitest`:

```
test 43
        Start  43: UnitStateMagicTest.schoolLevelByDefault

43: Test command: /home/runner/work/vcmi/vcmi/out/build/linux-gcc-test/bin/vcmitest "--gtest_filter=UnitStateMagicTest.schoolLevelByDefault" "--gtest_also_run_disabled_tests"
43: Working Directory: /home/runner/work/vcmi/vcmi/out/build/linux-gcc-test/bin/
43: Test timeout computed to be: 10000000
43: Note: Google Test filter = UnitStateMagicTest.schoolLevelByDefault
43: [==========] Running 1 test from 1 test suite.
43: [----------] Global test environment set-up.
43: [----------] 1 test from UnitStateMagicTest
43: [ RUN      ] UnitStateMagicTest.schoolLevelByDefault
43: unknown file: Failure
43: Uninteresting mock function call - returning default value.
43:     Function call: getId()
43:           Returns: -1
43: [  FAILED  ] UnitStateMagicTest.schoolLevelByDefault (0 ms)
43: [----------] 1 test from UnitStateMagicTest (0 ms total)
43: 
43: [----------] Global test environment tear-down
43: Ending global test tear-down.
43: [==========] 1 test from 1 test suite ran. (213 ms total)
43: [  PASSED  ] 0 tests.
43: [  FAILED  ] 1 test, listed below:
43: [  FAILED  ] UnitStateMagicTest.schoolLevelByDefault
43: 
43:  1 FAILED TEST
 43/261 Test  #43: UnitStateMagicTest.schoolLevelByDefault ...............................................***Failed    0.22 sec
```

`set_tests_properties(NoBattleIsDraw PROPERTIES DISABLED TRUE)` doesn't work either:
https://stackoverflow.com/a/76274577/2278742
https://gitlab.kitware.com/cmake/cmake/-/issues/22813

**Comment by Alexander-Wilms** [comment]
Now the CI run is successful: https://github.com/Alexander-Wilms/vcmi/actions/runs/7393513222/job/20113545264

```
100% tests passed, 0 tests failed out of 180
Total Test time (real) =  42.83 sec
The following tests did not run:
	  8 - BattleFinishedTest.NoBattleIsDraw (Disabled)
	 43 - UnitStateMagicTest.schoolLevelByDefault (Disabled)
	 44 - UnitStateMagicTest.schoolLevelForNormalCaster (Disabled)
	 45 - UnitStateMagicTest.effectLevelForNormalCaster (Disabled)
	 51 - UnitStateMagicTest.effectValueByDefault (Disabled)
	 52 - UnitStateMagicTest.effectValue (Disabled)
	 64 - CCreatureTest.JsonUpdate (Disabled)
	 65 - CCreatureTest.JsonAddBonus (Disabled)
	 66 - CCreatureTest.JsonRemoveBonus (Disabled)
	 82 - CGameStateTest.issue2765 (Disabled)
	 83 - CGameStateTest.battleResurrection (Disabled)
	 87 - MapFormat.Random (Disabled)
	 88 - MapFormat.Objects (Disabled)
	 89 - MapFormat.Terrain (Disabled)
	113 - CatapultTest.ApplicableInTown (Disabled)
	114 - CatapultApplyTest.DamageToIntactPart (Disabled)
	118 - CloneApplyTest.AddsNewUnit (Disabled)
	119 - CloneApplyTest.SetsClonedFlag (Disabled)
	120 - CloneApplyTest.SetsCloneLink (Disabled)
	121 - CloneApplyTest.SetsLifetimeMarker (Disabled)
	124 - DamageApplyTest.DoesDamageToAliveUnit (Disabled)
	125 - DamageApplyTest.IgnoresDeadUnit (Disabled)
	126 - DamageApplyTest.DoesDamageByPercent (Disabled)
	127 - DamageApplyTest.DoesDamageByCount (Disabled)
	128 - DispelTest.ApplicableToAliveUnitWithTimedEffect (Disabled)
	129 - DispelTest.IgnoresOwnEffects (Disabled)
	132 - DispelApplyTest.RemovesEffects (Disabled)
	139 - HealTest.NotApplicableIfDeadUnitIsBlocked (Disabled)
	140 - HealTest.ApplicableWithAnotherDeadUnitInSamePosition (Disabled)
	143 - SacrificeApplyTest.ResurrectsTarget (Disabled)
	144 - TeleportApplyTest.MovesUnit (Disabled)
	152 - BonusConditionTest.ReceptiveIfMatchesType (Disabled)
	163 - LuaSandboxTest.Example (Disabled)
	167 - LuaSpellEffectAPITest.ApplicableOnExpert (Disabled)
	168 - LuaSpellEffectAPITest.NotApplicableOnAdvanced (Disabled)
	169 - LuaSpellEffectAPITest.ApplicableOnLeftSideOfField (Disabled)
	170 - LuaSpellEffectAPITest.NotApplicableOnRightSideOfField (Disabled)
	171 - LuaSpellEffectAPITest.ApplyMoveUnit (Disabled)
	177 - ByConfig1/HealApplyTest.Heals/(1-byte object <00>, 1-byte object <01>) (Disabled)
	178 - ByConfig1/HealApplyTest.Heals/(1-byte object <01>, 1-byte object <01>) (Disabled)
	179 - ByConfig1/HealApplyTest.Heals/(1-byte object <02>, 1-byte object <01>) (Disabled)
	180 - ByConfig2/HealApplyTest.Heals/(1-byte object <01>, 1-byte object <00>) (Disabled)
	181 - ByConfig2/HealApplyTest.Heals/(1-byte object <02>, 1-byte object <00>) (Disabled)
	182 - ByConfig/SummonTest.Applicable/(-1, false, false) (Disabled)
	183 - ByConfig/SummonTest.Applicable/(-1, false, true) (Disabled)
	184 - ByConfig/SummonTest.Applicable/(-1, true, false) (Disabled)
	185 - ByConfig/SummonTest.Applicable/(-1, true, true) (Disabled)
	186 - ByConfig/SummonTest.Applicable/(112, false, false) (Disabled)
	187 - ByConfig/SummonTest.Applicable/(112, false, true) (Disabled)
	188 - ByConfig/SummonTest.Applicable/(112, true, false) (Disabled)
	189 - ByConfig/SummonTest.Applicable/(112, true, true) (Disabled)
	190 - ByConfig/SummonTest.Applicable/(114, false, false) (Disabled)
	191 - ByConfig/SummonTest.Applicable/(114, false, true) (Disabled)
	192 - ByConfig/SummonTest.Applicable/(114, true, false) (Disabled)
	193 - ByConfig/SummonTest.Applicable/(114, true, true) (Disabled)
	194 - ByConfig/SummonTest.Transform/(-1, false, false) (Disabled)
	195 - ByConfig/SummonTest.Transform/(-1, false, true) (Disabled)
	196 - ByConfig/SummonTest.Transform/(-1, true, false) (Disabled)
	197 - ByConfig/SummonTest.Transform/(-1, true, true) (Disabled)
	198 - ByConfig/SummonTest.Transform/(112, false, false) (Disabled)
	199 - ByConfig/SummonTest.Transform/(112, false, true) (Disabled)
	200 - ByConfig/SummonTest.Transform/(112, true, false) (Disabled)
	201 - ByConfig/SummonTest.Transform/(112, true, true) (Disabled)
	202 - ByConfig/SummonTest.Transform/(114, false, false) (Disabled)
	203 - ByConfig/SummonTest.Transform/(114, false, true) (Disabled)
	204 - ByConfig/SummonTest.Transform/(114, true, false) (Disabled)
	205 - ByConfig/SummonTest.Transform/(114, true, true) (Disabled)
	206 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, false) (Disabled)
	207 - ByConfig/SummonApplyTest.SpawnsNewUnit/(false, true) (Disabled)
	208 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, false) (Disabled)
	209 - ByConfig/SummonApplyTest.SpawnsNewUnit/(true, true) (Disabled)
	210 - ByConfig/SummonApplyTest.UpdatesOldUnit/(false, false) (Disabled)
	211 - ByConfig/SummonApplyTest.UpdatesOldUnit/(false, true) (Disabled)
	212 - ByConfig/SummonApplyTest.UpdatesOldUnit/(true, false) (Disabled)
	213 - ByConfig/SummonApplyTest.UpdatesOldUnit/(true, true) (Disabled)
	214 - ByConfig/TimedApplyTest.ChangesBonuses/(false, false) (Disabled)
	215 - ByConfig/TimedApplyTest.ChangesBonuses/(false, true) (Disabled)
	216 - ByConfig/TimedApplyTest.ChangesBonuses/(true, false) (Disabled)
	217 - ByConfig/TimedApplyTest.ChangesBonuses/(true, true) (Disabled)
	218 - BySpells/AbsoluteSpellConditionTest.ChecksAbsoluteCase/(1, 1) (Disabled)
	219 - BySpells/AbsoluteSpellConditionTest.ChecksAbsoluteCase/(1, 2) (Disabled)
	220 - BySpells/AbsoluteSpellConditionTest.ChecksAbsoluteCase/(2, 1) (Disabled)
	221 - BySpells/AbsoluteSpellConditionTest.ChecksAbsoluteCase/(2, 2) (Disabled)
	222 - BySpells/AbsoluteSpellConditionTest.IgnoresNormalCase/(1, 1) (Disabled)
	223 - BySpells/AbsoluteSpellConditionTest.IgnoresNormalCase/(1, 2) (Disabled)
	224 - BySpells/AbsoluteSpellConditionTest.IgnoresNormalCase/(2, 1) (Disabled)
	225 - BySpells/AbsoluteSpellConditionTest.IgnoresNormalCase/(2, 2) (Disabled)
	258 - BySpells/NormalSpellConditionTest.ChecksAbsoluteCase/(1, 1) (Disabled)
	259 - BySpells/NormalSpellConditionTest.ChecksAbsoluteCase/(1, 2) (Disabled)
	260 - BySpells/NormalSpellConditionTest.ChecksAbsoluteCase/(2, 1) (Disabled)
	261 - BySpells/NormalSpellConditionTest.ChecksAbsoluteCase/(2, 2) (Disabled)
	262 - BySpells/NormalSpellConditionTest.ChecksNormalCase/(1, 1) (Disabled)
	263 - BySpells/NormalSpellConditionTest.ChecksNormalCase/(1, 2) (Disabled)
	264 - BySpells/NormalSpellConditionTest.ChecksNormalCase/(2, 1) (Disabled)
	265 - BySpells/NormalSpellConditionTest.ChecksNormalCase/(2, 2) (Disabled)
```

**Comment by nullkiller** [comment]
I created a repository here https://github.com/vcmi-mods/vcmi-test-data
The idea is to use password protected zip instead so that we can not even see what is inside it without github secret. We will put a zip as release output in this new repository. Need to add a readme there.

Are you ok with this solution? If it is hard to unzip with password we can encrypt zip file with python instead.

**Comment by nullkiller** [comment]
I can add readme myself. And will upload test data as well as soon as we decide upon approach

**Comment by nullkiller** [comment]
Lets start internal tests

**Pull request merged**